src/main/java/org/elasticsearch/common/util/concurrent/BaseFuture.java

```diff
@@ -118,6 +117,22 @@ public V get() throws InterruptedException, ExecutionException {
         return sync.get();
     }
 
+    /**
+     * Similar to {@link #get()} with the exception that it never blocks. If the operation wasn't completed yet
+     * it will return {@code null}. Otherwise it will behave in a similar fashion to {@link #get()} and will return
+     * the computed result, including throwing a {@link CancellationException} or {@link ExecutionException} if
+     * the computation was cancelled or threw an error (respectively).
+     *
+     * @throws CancellationException if the computation was cancelled
+     * @throws ExecutionException    if the computation threw an exception
+     */
+    public V tryGet() throws CancellationException, ExecutionException {
+        if (sync.isDone() == false) {
+            return null;
+        }
+        return sync.getValue();
+    }
```

Comment:
Returning `null` feels wrong to me as then there is no difference between tryGet if the computation is not done and if it actually returns `null`. Maybe we need different semantics such as throwing an exception instead of returning `null` is sync.isDone() returns false?